Spatial.gd

```diff
@@ -760,3 +861,39 @@ func _on_Settings_pressed():
 	settings_dialog.load_settings()
 	settings_dialog.show()
 
+
+func _on_MarkerPacks_cell_selected():
+	var category_item = marker_packs.get_selected()
+	self.currently_active_category = category_item.get_metadata(0)
+	for path in self.paths.get_children():
+		if path.category_name == find_pedigree_name(category_item, category_item.get_metadata(0).get_name()):
+			path.is_editable = true
+			self.currently_active_path = path
+		else:
+			path.is_editable = false
+	for icon in self.icons.get_children():
+		if icon.category_name == find_pedigree_name(category_item, category_item.get_metadata(0).get_name()):
+			icon.is_editable = true
+		else:
+			icon.is_editable = false
+
+
+func _on_MarkerPacks_item_edited():
+	var category_item = marker_packs.get_edited()
+	switch_selected_category(category_item)
+
+
+func _on_ShowAll_pressed():
+	var category_item = self.marker_packs.get_root().get_children()
+	while category_item != null:
+		category_item.set_checked(1, true)
+		switch_selected_category(category_item)
+		category_item = category_item.get_next()	
+
+
+func _on_HideAll_pressed():
+	var category_item = self.marker_packs.get_root().get_children()
+	while category_item != null:
+		category_item.set_checked(1, false)
+		switch_selected_category(category_item)
```

Comment:
This will probably need to change when we change the checkbox hierarchy ux

Reply:
As discussed, Show and Hide All have been removed